### PR TITLE
Add CoreFx to known-project list and move to tip of release/2.1 branch

### PR DESCRIPTION
**Pull request by weshaggard**
cc @dseefeld @dagood @crummel 

Using this PR to test and will work through any CI failures.

### CODE REVIEW

**Comment by weshaggard** [comment]
The corefx issue is caused by a change in how we consume ilasm (that is always causing us one issue or another). See https://github.com/dotnet/corefx/pull/25649 we are going to need to make similar changes in source-build to setup the tool correctly.

File: tools-local/ILAsmVersion.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+2.1.0-preview2-26314-17
```
Comment (dagood) [inline tools-local/ILAsmVersion.txt]: I think we should set up the updater for this: https://github.com/dotnet/corefx/blob/305bebcb8e9c4c0b05ecf6b2f87fa723f1a33493/dependencies.props#L183-L187

Comment (weshaggard) [reply tools-local/ILAsmVersion.txt]: good call I will do that.

File: init-tools.sh [hunk 2]
```diff
@@ -106,28 +108,32 @@ if [ ! -e "$__DOTNET_PATH" ]; then
...
     echo "Installing dotnet cli..."
     __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.tar.gz"
 
     install_dotnet_cli() {
-        echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> "$__init_tools_log"
-        rm -rf -- "$__DOTNET_PATH/*"
-        # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-        if command -v curl > /dev/null; then
-            curl --retry 10 -sSL --create-dirs -o "$__DOTNET_PATH/dotnet.tar" "${__DOTNET_LOCATION}"
+        if [[ -z "${DotNetBootstrapCliTarPath-}" ]]; then
```
Comment (dagood) [inline init-tools.sh]: Does this work? I only know of the ones listed here: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html, would expect `:-`.

Comment (omajid) [reply init-tools.sh]: See http://wiki.bash-hackers.org/syntax/pe#use_a_default_value

Comment (weshaggard) [reply init-tools.sh]: Yes it works as a test. You have to read the fine print:

```
When not performing substring expansion, using the form described below (e.g., ‘:-’), Bash tests for a parameter that is unset or null. Omitting the colon results in a test only for a parameter that is unset. Put another way, if the colon is included, the operator tests for both parameter’s existence and that its value is not null; if the colon is omitted, the operator tests only for existence.
```

Also I like http://wiki.bash-hackers.org/syntax/pe#use_a_default_value as a reference, little easier to read.

Comment (weshaggard) [reply init-tools.sh]: For what it is worth while looking this up there are about 100 different ways to do this check so I'm not sure there is a defacto way but I'm happy to be pointed to a clear guide. The best answer I found was https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash and I couldn't figure out why I would use `+` over `-` so I just stuck with what we already have in this script.

Comment (dagood) [reply init-tools.sh]: Oh, wow, I've been missing that for a while. Thanks for the explanation, and yeah, that wiki is nicer.

**Comment by weshaggard** [comment]
```

/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/corefx/external/runtime/runtime.depproj : error NU1103: Unable to find a stable package Microsoft.NETCore.TestHost with version (>= 1.1.2) [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/corefx.proj]
15:17:51 /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/corefx/external/runtime/runtime.depproj : error NU1103: - Found 1 version(s) in /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/prebuilt/nuget-packages/ [ Nearest version: 2.1.0-preview2-26314-17 ] [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/corefx.proj]
15:17:51 /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/corefx/external/runtime/runtime.depproj : error NU1103: - Found 0 version(s) in /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/bin/obj/x64/Release/blob-feed/packages/ [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/corefx.proj
```

@crummel @dseefeld any ideas why this tarball build might be failing in this way? I don't even see anywhere in corefx were we reference the stable 1.1.2 version of package. We are referencing the version of the CoreCLR we just built in that project so I'm not sure what is going on.

**Comment by dseefeld** [comment]
@weshaggard here's what I'm seeing:

**prebuilt** has:

CoreCLR:
- microsoft.netcore.runtime.coreclr.1.0.5.nupkg
- microsoft.netcore.runtime.coreclr.1.1.1.nupkg
- microsoft.netcore.runtime.coreclr.1.1.2.nupkg
- microsoft.netcore.runtime.coreclr.2.1.0-preview2-26314-17.nupkg
- runtime.linux-x64.microsoft.netcore.runtime.coreclr.2.1.0-preview2-26314-17.nupkg

 TestHost:
- microsoft.netcore.testhost.2.1.0-preview2-26314-17.nupkg
- runtime.linux-x64.microsoft.netcore.testhost.2.1.0-preview2-26314-17.nupkg

**blob-feed** has:

CoreCLR:
- Microsoft.NETCore.Runtime.CoreCLR.1.0.2.nupkg
- Microsoft.NETCore.Runtime.CoreCLR.1.0.4.nupkg
- Microsoft.NETCore.Runtime.CoreCLR.1.0.5.nupkg
- Microsoft.NETCore.Runtime.CoreCLR.1.0.7.nupkg
- Microsoft.NETCore.Runtime.CoreCLR.1.1.0.nupkg
- Microsoft.NETCore.Runtime.CoreCLR.1.1.1.nupkg
- Microsoft.NETCore.Runtime.CoreCLR.1.1.2.nupkg
- runtime.centos.7-x64.Microsoft.NETCore.Runtime.CoreCLR.2.1.0-preview2-25519-03.nupkg
- runtime.centos.7-x64.Microsoft.NETCore.Runtime.CoreCLR.2.1.0-preview2-25519-03.nupkg.requires_nupkg_signing
- runtime.osx.10.10-x64.Microsoft.NETCore.Runtime.CoreCLR.1.1.0.nupkg
- runtime.ubuntu.14.04-x64.Microsoft.NETCore.Runtime.CoreCLR.1.1.0.nupkg
- runtime.ubuntu.16.04-x64.Microsoft.NETCore.Runtime.CoreCLR.1.1.0.nupkg
- runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR.1.0.7.nupkg
- runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR.1.1.0.nupkg
- transport.runtime.centos.7-x64.Microsoft.NETCore.Runtime.CoreCLR.2.1.0-preview2-25519-03.nupkg
 
TestHost:
- runtime.centos.7-x64.Microsoft.NETCore.TestHost.2.1.0-preview2-25519-03.nupkg
- runtime.centos.7-x64.Microsoft.NETCore.TestHost.2.1.0-preview2-25519-03.nupkg.requires_nupkg_signing
- transport.runtime.centos.7-x64.Microsoft.NETCore.TestHost.2.1.0-preview2-25519-03.nupkg

**PackageVersion.props** has:

CoreCLR:
**`<MicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.1.2</MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`**
`<RuntimeCentos7X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>2.1.0-preview2-25519-03</RuntimeCentos7X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`
`<RuntimeOsx1010X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.1.0</RuntimeOsx1010X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`
`<RuntimeUbuntu1404X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.1.0</RuntimeUbuntu1404X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`
`<RuntimeUbuntu1604X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.1.0</RuntimeUbuntu1604X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`
`<RuntimeWin7X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.1.0</RuntimeWin7X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`
`<TransportRuntimeCentos7X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>2.1.0-preview2-25519-03</TransportRuntimeCentos7X64MicrosoftNETCoreRuntimeCoreCLRPackageVersion>`

TestHost:
`<RuntimeCentos7X64MicrosoftNETCoreTestHostPackageVersion>2.1.0-preview2-25519-03</RuntimeCentos7X64MicrosoftNETCoreTestHostPackageVersion>`
`<TransportRuntimeCentos7X64MicrosoftNETCoreTestHostPackageVersion>2.1.0-preview2-25519-03</TransportRuntimeCentos7X64MicrosoftNETCoreTestHostPackageVersion>`

https://github.com/dotnet/corefx/blob/2c0a662262a6ad8bfdfb97dd786db2734c24754b/external/runtime/runtime.depproj#L34 is using the package version for Runtime.CoreCLR (1.1.2) and there's not a TestHost package in prebuilt or blob-feed with this version.

**Comment by weshaggard** [comment]
```
<MicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.1.2</MicrosoftNETCoreRuntimeCoreCLRPackageVersion>
```

That is wrong we need to understand why it is picking 1.1.2 version of the runtime package.

**Comment by dagood** [comment]
It looks to me like the CoreCLR is only producing runtime packages, not the bait and switch ones (forget the technical term for those) so the code that generates the package version props doesn't have a newer one to pick up than 1.1.2.

For example, in CoreCLR official builds, they're produced by the Windows build:

```
Microsoft.NETCore.TestHost -> E:\A\_work\37\s\bin\Product\Windows_NT.x64.Release\.nuget\/pkg/specs/runtime.win-x64.Microsoft.NETCore.TestHost.nuspec
Microsoft.NETCore.TestHost -> E:\A\_work\37\s\bin\Product\Windows_NT.x64.Release\.nuget\/pkg/specs/Microsoft.NETCore.TestHost.nuspec
```

In the logs here (and in official linux builds) only the runtime packages are produced:

```
Microsoft.NETCore.TestHost -> /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/bin/obj/x64/Release/blob-feed/packages/specs/runtime.rhel.7-x64.Microsoft.NETCore.TestHost.nuspec
```

**Comment by weshaggard** [comment]
I believe I tracked down why the identity packages aren't being built. See https://github.com/dotnet/coreclr/blob/release/2.1/src/.nuget/dir.traversal.targets#L6. It is because when OfficialBuildId is set we only build them for a specific configuration to avoid duplication. 

@dseefeld we are going to need to make a change similar to https://github.com/dotnet/corefx/pull/25586 in both coreclr and corefx. I will work on that.

File: repos/dir.targets [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <EnvironmentVariables Include="OfficialBuildId=$(OfficialBuildId)" />
     <EnvironmentVariables Include="VersionSeedDate=$(OfficialBuildId)" />
     <!-- Work around a dependency on .git that doesn't have a good fallback yet: https://github.com/dotnet/buildtools/issues/1980 -->
-    <EnvironmentVariables Include="GitBranchName=N/A" />
+    <EnvironmentVariables Include="RawGitBranchName=N/A" />
```
Comment (dagood) [inline repos/dir.targets]: Can remove this workaround completely, as discussed in https://github.com/dotnet/buildtools/issues/1980.

**Comment by weshaggard** [comment]
@dagood @dseefeld would you guys mind taking a look at the failure to see if you can help me narrow down how the XUnitPackageVersion is getting set to 2.2.0 but only in the tarball build.

**Comment by weshaggard** [comment]
>@dagood @dseefeld would you guys mind taking a look at the failure to see if you can help me narrow down how the XUnitPackageVersion is getting set to 2.2.0 but only in the tarball build.

Nevermind I think I just figured it out, it is because of https://github.com/dotnet/source-build/blob/dev/release/2.1/tools-local/init-build.proj#L275 and we are sharing the version https://github.com/dotnet/corefx/blob/release/2.1/dependencies.props#L231. This should have never worked. For now I'm going to fix it by adding xunit.console.runner to the dummy package list.

**Comment by weshaggard** [comment]
Finally a green build... now to move to the new tip of corefx/coreclr release/2.1 branches and make sure things still work :)

**Comment by weshaggard** [comment]
OK this is now ready, when CI is green we can merge.
